model/demo_model.py

```diff
@@ -0,0 +1,208 @@
+            self.currently_active_lamps = self.current_active_lamps_list[current_time_in_minutes]
+            
+            prev_time_in_minutes = self._get_minutes_from_datetime(self.prev_time)
+            prev_active_lamps = self.current_active_lamps_list[prev_time_in_minutes]
+            
+            lamps_to_update = []
+            if self.current_time.minute > self.prev_time.minute and prev_active_lamps != self.currently_active_lamps:
+                
+                lamps_to_update = list(set(prev_active_lamps).symmetric_difference(set(self.current_active_lamps_list[current_time_in_minutes])))
+                print(F"{lamps_to_update} are to be updated")
+                
+                if len(lamps_to_update) == 0:
+                    for lamp in prev_active_lamps:
+                        self.publish_lamp_event(self.current_time,lamp,LampAction.OFF)
+                        
+                for lamp in lamps_to_update:
+                    # If lamp was part of active lamps then it should turn off, else on
+                    if lamp in prev_active_lamps:
+                        self.publish_lamp_event(self.current_time,lamp,LampAction.OFF)
+                    else:
+                        self.publish_lamp_event(self.current_time,lamp,LampAction.ON)
+                self.prev_time = self.current_time
+
+    def _get_minutes_from_datetime(self, time : datetime):
+        return time.hour * 60 + time.minute
+
+    def get_schedule(self):
+        user_data = "tools\mock_user_data.json"
+        user_events = self.read_data(user_data)
+        return self.scheduler.createSchedule(user_events)
+    
+    def get_current_active_lights(self):
+        return self.currently_active_lamps
+
+    def get_user_schedule(self):
+        """Get user data from a single day.
+        """
+        user_data = "tools\mock_user_data.json"
+        user_events = self.read_data(user_data)
+        date_schedule_is_based_on = user_events[0].timestamp.date()
+        day_specific_events = []
+        for events in user_events:
+            if events.timestamp.date() == date_schedule_is_based_on:
+                day_specific_events.append(events)
+            else:
+                break
+            
+            
+        user_schedule = Schedule.createSchedule(day_specific_events)     
+        return user_schedule
+    
+    def create_active_lamp_list(self) -> List[List[str]]:
+        events = self.current_schedule.events[:]
+        
+        active_lamps = [[] for _ in range(1440)]
+        
+        checked_lamps = []
+        
+        for e in events:
+            lamp_name = e.lamp
+            for e2 in events:
+                event_timestamp = e2.timestamp.hour * 60 + e2.timestamp.minute
+                if e2.lamp == lamp_name and not lamp_name not in checked_lamps:
+                    for i in range(event_timestamp,1440):
+                        if e2.action.value == "on" and lamp_name not in active_lamps[i]:
```

Comment:
Could we use the enums here instead of `"on"` and `"off"`?

Reply:
yes, oversight from my part